benchmark/result/result.py

```diff
@@ -0,0 +1,48 @@
+"""Basic result classes for benchmarking."""
+
+from typing import Any
+from uuid import UUID
+
+from attrs import define, field
+from attrs.validators import instance_of
+from pandas import DataFrame
+
+
+@define(frozen=True)
+class Result:
+    """A single result of the benchmarking."""
+
+    name: str = field(validator=instance_of(str))
+    """The name of the benchmarking result."""
+
+    identifier: UUID = field(validator=instance_of(UUID))
```

Comment:
If you execute a `Benchmark`, the created `Result` will get the exact same UUID. This is not desirable: Even when executing the same `Benchmark` twice, it could happen that I get e.g. different execution times due to numerical/rounding issues. Thus, this identifier should not just be the one that you get from the `Benchmark`.
Also, is it possible to just have this created automatically upon initialization?